libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -284,6 +287,12 @@ string ABIFunctions::conversionFunction(Type const& _from, Type const& _to)
 		auto fromCategory = _from.category();
 		switch (fromCategory)
 		{
+		case Type::Category::Address:
+			body =
+				Whiskers("converted := <convert>(value)")
+					("convert", conversionFunction(IntegerType(160), _to))
```

Comment:
Why is this considering address as `IntegerType(160)` ?

Reply:
It calls the conversion function for uint160 because the binary representation is identical.

Reply:
Yeah but I think this is the wrong approach. The `conversionFunction` should just accept the address type and then make a decision itself what the representation is.